### DISCUSSION TITLE
analize a can trace

### DISCUSSION THREAD

**Discussion by MCO-SOCOMEC** [canopen-python/canopen#330] [Ideas]
Hi,
I saw your project and I'm wondering if it's possible to use it to process a can trace an analize it to get PDO transmitted, SDO, .. Would it be possible for you? I'm particularly interested in SDO decoding and analysis (to understand if there are continuous request to an object, ..). I looked in the web but I was not able to find something doing that.
Thanks
michele

**Comment by christiansandberg** [comment]
Unfortunately it was not designed to do that. It is very bound to its own I/O. It would require quite a large rewrite but it could definitely be beneficial to separate processing data from the actual I/O.

**Reply by MCO-SOCOMEC** [reply]
thank you very much for your reply.
Michele

**Comment by klausz** [comment]
To avoid misunderstandings and i am also not 100% if i got it right.
In using CANplayer:
`canplayer vcan0=can0 -I abc.log -g 1000`
it is possible to play/repeat a CANopenDump.log file. 

With Jupyter notebooks and the code:
```
import canopen

network = canopen.Network()

node = network.add_node(0x47, 'ab.eds') 
node2 = network.add_node(01a, 'cd.eds') 

network.connect(bustype='socketcan', channel='vcan0')

for obj in node2.object_dictionary.values():
    print('0x%X: %s' % (obj.index, obj.name))
    if isinstance(obj, canopen.objectdictionary.Record):
        for subobj in obj.values():
            print('  %d: %s' % (subobj.subindex, subobj.name))
```
_0x1000: DeviceType
0x1001: ErrorRegister
0x1002: Manufacturer status register_

`node.object_dictionary[0x2001][1].__dict__`

_{'parent': <canopen.objectdictionary.Array at 0x7f4b2f479300>,
 'index': 8193,
 'subindex': 1,
..False}_

`node.object_dictionary[0x2021][5].default_raw`
_'440'_

I can get a lot of information. 
What i miss, 
(1) is it possible to get a online translation of the CANopen and .eds content (mapped) and displayed in the terminal window while CAN (real or CANplayer) is running? A terminal command in Linux displaying CANopen data thats addionaly calling CANOPEN and provides  the link to the .eds file (one would be enough).
(2) Also i am not sure - maybe i just need to make the code - to get the raw data from all the nodes when i start the cell in Jupyter?

**Comment by smita6789** [comment]
Is there any way to create log file for CAN trace with python canopen library? Like text or any other file with data as timestamp, CAN ID, request, response etc
